WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModelProtocol.swift

```diff
@@ -180,4 +180,11 @@ extension ProductFormViewModelProtocol {
             }
         }
     }
+
+    /// Whether the Preview button should be enabled, when it's available in the navigation bar.
+    /// Returns `false` when it's a new blank product without any changes.
+    ///
+    func shouldEnablePreviewButton() -> Bool {
+        !(formType == .add && !hasUnsavedChanges())
+    }
```

Comment:
Good idea about toggling the button state instead of hiding it!

Reply:
We may want to change this condition due to the template experiment, as a new template product would be considered as a new product without any changes. 

I think we need to add a check for the "blank" state.

Reply:
> I think we need to add a check for the "blank" state.

I agree, but I haven't come up with a good solution for it yet. This PR doesn't change whether you can preview new template products (it just makes the preview button visible but disabled, instead of not appearing at all), so I'm going to go ahead and merge it. I've opened a new issue https://github.com/woocommerce/woocommerce-ios/issues/8071 so we can follow up with a fix for new template products.